op-node/sources/caching/blob_lru.go

```diff
@@ -0,0 +1,82 @@
+package caching
+
+import (
+	lru "github.com/hashicorp/golang-lru/v2"
+	"math"
+	"sync"
+)
+
+type SizeFn func(value any) int
+
+// SizeConstrainedCache is a cache where capacity is in bytes (instead of item count). When the cache
+// is at capacity, and a new item is added, older items are evicted until the size
+// constraint is met.
+//
+// OBS: This cache assumes that items are content-addressed: keys are unique per content.
+// In other words: two Add(..) with the same key K, will always have the same value V.
+type SizeConstrainedCache[K comparable, V any] struct {
+	m       Metrics
+	label   string
+	size    int
```

Comment:
Perhaps uint64 type should be used instead, it is uncertain if int type will overflow when counting byte size.